### ISSUE TITLE
Wrong result for Table.getQualifiedName() on aliased table(Name)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12307]
```java
System.out.println(table(name("S", "T")).as("A").getQualifiedName());
```

Produces

```
"S"."A"
```

This issue causes, among other things: https://github.com/jOOQ/jOOQ/issues/12306

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0 and 3.15.2 (#12311)

**Comment by lukaseder** [comment]
The same thing needs to be applied to `field(Name)`
